fix: dropdown widget fetching output files

LittleSound opened the pull request:
related https://github.com/Comfy-Org/ComfyUI_frontend/issues/5827#issuecomment-3539629932

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-6734-fix-dropdown-widget-fetching-output-files-2af6d73d365081988340f74694b7cee7) by [Unito](https://www.unito.io)

viva-jinyi reviewed (changes requested):
It doesn’t seem to cover both Cloud and OSS. Since `internal/files` can only fetch output files in the OSS environment, I think it would be better to use `useMediaAssets.ts`. It’s already implemented to work for both OSS and Cloud, and we can update it later when the new API becomes available.

viva-jinyi commented:
@LittleSound Can you resolve conflicts?

viva-jinyi reviewed (dismissed):
LGTM

christian-byrne committed:
fix: use useMediaAssets for dropdown output files instead of queueStore

Replace queueStore.historyTasks approach with useMediaAssets('output')
for fetching output files in dropdown widgets. This works for both
Cloud and OSS environments.

- Use getMediaTypeFromFilename for consistent media type detection
- Convert FormDropdown isOpen from ref to defineModel for parent access
- Refresh output assets on dropdown open
- Add audio support to output file filtering

Fixes #5827

christian-byrne reviewed (approved):
LGTM
